src/mlpack/methods/ann/layer/transposed_convolution_impl.hpp

```diff
@@ -407,6 +428,43 @@ void TransposedConvolution<
     aH = (outputHeight + kernelHeight - 2 * padHeight - 2) % strideHeight;
   }
 }
+template<
+    typename ForwardConvolutionRule,
+    typename BackwardConvolutionRule,
+    typename GradientConvolutionRule,
+    typename InputDataType,
+    typename OutputDataType
+>
+void TransposedConvolution<
+    ForwardConvolutionRule,
+    BackwardConvolutionRule,
+    GradientConvolutionRule,
+    InputDataType,
+    OutputDataType
+>::InitializeSamePadding(){
+  /*
+  * Using O=s*(I-1) + K -2P + A
+  * where
+  * s=stride 
+  * I=Input Shape
+  * K=Kernel Size
+  * P=Padding
+  */
+
+  padWidth = (kernelWidth-strideWidth)/2 + kernelWidth-strideWidth%2 == 1;
+  padHeight = (kernelHeight-strideHeight)/2 + kernelHeight-strideHeight%2 == 1;
+  // If Padding is negative set it to 0.
```

Comment:
Hm, the padding can be < 0 if the user passed in valid parameters? If not, instead of setting  `padWidth = 0`, we should print out a failure and exit.

Reply:
With Same padding if stride width < kernel width then calculated padWidth will be negative which is not possible so if such a case is not possible there is already "Output dimension not possible " log.

Reply:
Thanks for the clarification, in this case let's add another fatal error here.

Reply:
I have added fatal error instead of setting it to 0. Please have a look.